src/main/java/io/reactivex/internal/schedulers/SchedulerPoolFactory.java

```diff
@@ -81,61 +82,77 @@ public static void start() {
         if (exec != null) {
             exec.shutdownNow();
         }
         POOLS.clear();
     }
 
     static {
-        boolean purgeEnable = true;
-        int purgePeriod = 1;
-
         Properties properties = System.getProperties();
 
-        if (properties.containsKey(PURGE_ENABLED_KEY)) {
-            purgeEnable = Boolean.getBoolean(PURGE_ENABLED_KEY);
-        }
+        PurgeProperties pp = new PurgeProperties();
+        pp.load(properties);
 
-        if (purgeEnable && properties.containsKey(PURGE_PERIOD_SECONDS_KEY)) {
-            purgePeriod = Integer.getInteger(PURGE_PERIOD_SECONDS_KEY, purgePeriod);
-        }
-
-        PURGE_ENABLED = purgeEnable;
-        PURGE_PERIOD_SECONDS = purgePeriod;
+        PURGE_ENABLED = pp.purgeEnable;
+        PURGE_PERIOD_SECONDS = pp.purgePeriod;
 
         start();
     }
 
+    static final class PurgeProperties {
+
+        boolean purgeEnable;
+
+        int purgePeriod;
+
+        void load(Properties properties) {
+            if (properties.containsKey(PURGE_ENABLED_KEY)) {
+                purgeEnable = Boolean.parseBoolean(properties.getProperty(PURGE_ENABLED_KEY));
+            } else {
+                purgeEnable = true;
+            }
+
+            if (purgeEnable && properties.containsKey(PURGE_PERIOD_SECONDS_KEY)) {
+                try {
+                    purgePeriod = Integer.parseInt(properties.getProperty(PURGE_PERIOD_SECONDS_KEY));
+                } catch (NumberFormatException ex) {
+                    purgePeriod = 1;
+                }
+            } else {
+                purgePeriod = 1;
+            }
+        }
+    }
+
     /**
      * Creates a ScheduledExecutorService with the given factory.
      * @param factory the thread factory
      * @return the ScheduledExecutorService
      */
     public static ScheduledExecutorService create(ThreadFactory factory) {
         final ScheduledExecutorService exec = Executors.newScheduledThreadPool(1, factory);
-        if (PURGE_ENABLED && exec instanceof ScheduledThreadPoolExecutor) {
+        tryPutIntoPool(PURGE_ENABLED, exec);
+        return exec;
+    }
+
+    static void tryPutIntoPool(boolean purgeEnabled, ScheduledExecutorService exec) {
+        if (purgeEnabled && exec instanceof ScheduledThreadPoolExecutor) {
             ScheduledThreadPoolExecutor e = (ScheduledThreadPoolExecutor) exec;
             POOLS.put(e, exec);
         }
-        return exec;
     }
 
     static final class ScheduledTask implements Runnable {
         @Override
         public void run() {
-            try {
-                for (ScheduledThreadPoolExecutor e : new ArrayList<ScheduledThreadPoolExecutor>(POOLS.keySet())) {
-                    if (e.isShutdown()) {
-                        POOLS.remove(e);
-                    } else {
-                        e.purge();
-                    }
+            for (ScheduledThreadPoolExecutor e : new ArrayList<ScheduledThreadPoolExecutor>(POOLS.keySet())) {
```

Comment:
Removed the surrounding try-catch as it is practically impossible this loop crashes.